### PR TITLE
Add a card for access to multigpu partition

### PR DESCRIPTION
**Pull request by arsalanakhter**

### CODE REVIEW

File: docs/source/glossary.rst [hunk 1]
```diff
@@ -85,6 +88,9 @@ This glossary provides definitions for terms and abbreviations you may encounter
    Node Allocation
       The process of reserving a set of nodes for a specific job, ensuring that the required resources are available for successful execution.
 
+   Non-interactive job
+      A non-interactive job is a task submitted to a compute cluster that runs without the need for real-time user input. These jobs are defined by scripts that specify all the parameters and commands required for execution and do not provide a direct interface for user interaction during execution.**
```
Comment (joseph-cooke) [inline docs/source/glossary.rst:92]: I would just remove the ** and everything else looks okay. I might make mention that these are sbatch jobs just so people make the connection between a non-interactive job and the sbatch command.

File: docs/source/gpus/multigpu-partition-access.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Access to Multi-GPU Partition
+The `multigpu` partition in the HPC cluster allows users for extensive parallel processing. This setup is ideal for applications that require significant computational power, such as deep learning, scientific simulations, and large-scale data analysis.
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:2]: extensive parallel processing on multiple gpus.

File: docs/source/gpus/multigpu-partition-access.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+## Applying for Multi GPU reservation
+
+### Step 1
+
+- Use the partition application form link in here: [HPC Partition Request](https://bit.ly/NURC-PartitionAccess)
+- Select under "Partition Type" - `Multigpu - Partition (Testing Access)`
+- Fill in the number of GPUs and the type of GPU you'd like to test on. Please also provide a short description of your expected testing workload.
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:10]: I would add a Note somewhere on this step. (The testing has to be representative of the work that you are planning on completing but it does not have to be the full complete runs to get the timing data).

We should also push for testing on v100s if the jobs will fit scaled down so we are not allocating a100s for multigpu testing. If the job scales on a v100, it should scale on an a100 as well.

File: docs/source/gpus/multigpu-partition-access.md [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+- Submit the form.
+- RC will then contact you with further details for accessing a multi-gpu node. RC team will need an estimate of the duration needed for testing (preferably less than 24 hrs.)
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:12]: RC -> The Research Computing team

I would just expand this and not use the abbreviation because not sure if everyone will know what RC is if they just jump to this page.

File: docs/source/gpus/multigpu-partition-access.md [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+```
+
+***Note: Your local machine should be active for successful execution of the Job***
+
+- **To run {term}`Non-interactive job`, you can use the following command**
+    
+    1- Create a script
+    
+    ```bash
+    nano my_job.sh
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:37]: I might make a comment here there you can use the text editor of your choice. Just so no one thinks they have to use nano.

File: docs/source/gpus/multigpu-partition-access.md [hunk 6]
```diff
@@ -0,0 +1,94 @@
...
+    ```bash
+    #!/bin/bash
+    #SBATCH --job-name=MyJob             # Job name
+    #SBATCH -p reservation
+    #SBATCH --reservation=<reservation_name>
+    #SBATCH --gres=gpu:v100-sxm2:4
+    #SBATCH --nodes=1                    # Number of nodes
+    #SBATCH --ntasks=4                   # Number of tasks
+    #SBATCH --cpus-per-task=2            # Number of CPUs per task
+    #SBATCH --mem=8G                     # Total memory
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:51]: Make a comment that this is CPU memory and not GPU memory

File: docs/source/gpus/multigpu-partition-access.md [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+    #SBATCH --time=02:00:00              # Time limit hh:mm:ss
+    #SBATCH --output=output_%j.txt       # Standard output and error log
+    #SBATCH --error=error_%j.txt         # Standard error log
+    
+    # Load any required modules
+    module load python #example
+    
+    # Run your program
+    srun python your_script.py #example
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:60]: I would remove the srun statement. It really isn't needed in an sbatch script.

File: docs/source/gpus/multigpu-partition-access.md [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+    4-**You can monitor the status of your job using the squeue command:**
+    
+    ```bash
+    squeue -u <your_username>
+    
+    #To cancel a job
+    scancel <job_id> # where <job_id> is the ID of the Job we are cancelling
+    ```
+    
+- **You will also need to state whether it is for a class or for your own research work. If it is for a class, then the request should come from the instructor directly.**
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:83]: I would rephrase this to say this is meant for research workflows only. If a course needs multigpu, we would handle that workflow in the course form.

File: docs/source/gpus/multigpu-partition-access.md [hunk 9]
```diff
@@ -0,0 +1,94 @@
...
+    
+    ```bash
+    squeue -u <your_username>
+    
+    #To cancel a job
+    scancel <job_id> # where <job_id> is the ID of the Job we are cancelling
+    ```
+    
+- **You will also need to state whether it is for a class or for your own research work. If it is for a class, then the request should come from the instructor directly.** 
+RC staff should make sure that the instructor is contacted and educated about this and then create a reservation for all the students for the duration of the class.
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:84]: I think I might exclude this bit.

File: docs/source/gpus/multigpu-partition-access.md [hunk 10]
```diff
@@ -0,0 +1,94 @@
...
+- Perform testing on 1,2,..(4,8) GPUs. Record runtimes for each test (calculate efficiency).
+
+### Step-3
+
+**Post-Testing Application**
+
+- Ensure your {term}`Scaling efficiency` is adequate (generally over 0.5) when using the maximum GPUs selected on the `multigpu` partition⁠. If needed, please consult with the Research Computing (RC) team for guidance and support throughout the process⁠.
+- Re-enter the partition application form and select "Multigpu - Partition (Post Testing)" under "Partition Type."
```
Comment (joseph-cooke) [inline docs/source/gpus/multigpu-partition-access.md:92]: I would include the link for the ticket again.

File: docs/source/glossary.rst (hunk 1 above)
Comment (Rakshith2605) [inline docs/source/glossary.rst:92]:
```suggestion
      A non-interactive job (sbatch job) is a task submitted to a compute cluster that runs without the need for real-time user input. These jobs are defined by scripts that specify all the parameters and commands required for execution and do not provide a direct interface for user interaction during execution.
```

**Review by Rakshith2605 — APPROVED** [review]

File: docs/source/gpus/multigpu-partition-access.md (hunk 3 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:10]:
```suggestion
- Fill in the number of GPUs and the type of GPU you'd like to test on. Please also provide a short description of your expected testing workload.
.. note::
   **Important considerations while requesting GPU**
   - Prefer V100 GPUs for multi-GPU testing if your jobs fit:
     * V100 testing is often sufficient to demonstrate scaling.
     * If a job scales well on V100s, it's likely to scale similarly on A100s.
     * This approach conserves A100 resources for production work.
   - Testing should represent your planned work but doesn't need full production runs.
   - Use scaled-down versions of your jobs for timing data when possible.
   - Only use A100s for testing if your job specifically requires their capabilities or memory.
   - Ensure your test cases cover a range of processor/GPU counts to measure scaling accurately.
   - Consider testing with different problem sizes to understand how scaling efficiency changes with workload.

File: docs/source/gpus/multigpu-partition-access.md (hunk 4 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:12]:
```suggestion
- The Research Computing team will then contact you with further details for accessing a multi-GPU node. The Research Computing team will need an estimate of the duration needed for testing (preferably less than 24 hrs.)
```

File: docs/source/gpus/multigpu-partition-access.md [hunk 11]
```diff
@@ -0,0 +1,94 @@
...
+    #<reservation_name> will be in the details provided by the RC team.
+    ```
+    
+- To run an {term}`Interactive Job`, for example, using V100-sxm2 with 4 GPUs, use the command below
+
+```bash
+srun -p reservation --reservation=<reservation_name> --gres=gpu:v100-sxm2:4 --time=24:00:00 -N 1 --pty /bin/bash
+```
+
+***Note: Your local machine should be active for successful execution of the Job***
```
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:30]:
```suggestion
***Note***
   Your local machine must remain active for successful job execution. Be aware of the following risks:

   - Network disconnections will interrupt the job
   - Computer sleep/hibernation will break the connection
   - Losing RDP session (e.g., timeout, local reboot) stops GUI-dependent processes
   - Power outages or system updates can cause unexpected disconnects

   Recommendations:
   - Use a stable network connection
   - Disable sleep mode on your local machine
   - Consider using persistent remote desktop solutions
   - Implement job checkpointing where possible
   - Monitor job status regularly
```

File: docs/source/gpus/multigpu-partition-access.md (hunk 5 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:37]:
```suggestion
    nano my_job.sh 
    # You can use any text editor you prefer, not just nano. Choose the editor you're most comfortable with for modifying files. 
```

File: docs/source/gpus/multigpu-partition-access.md (hunk 6 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:51]:
```suggestion
    #SBATCH --mem=8G                     # Total CPU memory (not GPU memory)
```

File: docs/source/gpus/multigpu-partition-access.md (hunk 8 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:83]:
```suggestion
- **This multi-GPU setup is intended for research workflows only. For course-related multi-GPU needs, please refer to the course request form. Instructors should submit those requests directly through the appropriate channels.** 
```

File: docs/source/gpus/multigpu-partition-access.md (hunk 9 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:84]:
```suggestion

```

File: docs/source/gpus/multigpu-partition-access.md (hunk 10 above)
Comment (Rakshith2605) [inline docs/source/gpus/multigpu-partition-access.md:92]:
```suggestion
- Re-enter the partition application [form](https://bit.ly/NURC-PartitionAccess) and select "Multigpu - Partition (Post Testing)" under "Partition Type."
```

**Review by Rakshith2605 — APPROVED** [review]
